### PR TITLE
added abbility to distinquish between purchase and restore | added subscriptions support

### LINKED ISSUE #1: Add ability to distinguish between purchase vs restore
At the moment it is not possible to tell if something was purchased or restored.

Please add a way for the API to indicate which it is so that analytics can tell the difference, plus possibly useful in the future.

### PR DESCRIPTION
**Pull request by yurii-diachenko**
Fixes #1 and adds subscription support for Google

### CODE REVIEW

File: library/src/main/java/com/billing/BillingService.java [hunk 1]
```diff
@@ -23,11 +23,35 @@ public void removeListener(BillingServiceListener billingServiceListener) {
...
     }
 
-    public void productOwned(String sku) {
+    /**
+     * @param sku       - product specificator
+     * @param isRestore - a flag indicating whether it's a fresh purchase or restored product
+     */
+    public void productOwned(String sku, boolean isRestore) {
         Iterator<BillingServiceListener> iterator = billingServiceListeners.iterator();
         while (iterator.hasNext()) {
```
Comment (Apisov) [inline library/src/main/java/com/billing/BillingService.java:32]: I think we can replace all `while` loops with `foreach`, just for goodliness :)

File: library/src/main/java/com/billing/amazon/AmazonBillingListener.java [hunk 2]
```diff
@@ -74,15 +74,15 @@ public void onPurchaseResponse(PurchaseResponse response) {
                     Log.d(TAG, "onPurchaseResponse: receipt json:" + receipt.toJSON());
                     Log.d(TAG, "onPurchaseResponse: getUserId:" + response.getUserData().getUserId());
                     Log.d(TAG, "onPurchaseResponse: getMarketplace:" + response.getUserData().getMarketplace());
```
Comment (AlexDibrivnyy) [inline library/src/main/java/com/billing/amazon/AmazonBillingListener.java:76]: What do you think, is it good idea to log debug messages only if BuildConfig.DEBUG is true?

**Comment by mikemee** [comment]
@yurii-diachenko please add some more information about subscriptions to the readme. I started, but it could use some more. Thanks!

File: library/src/main/java/com/billing/amazon/AmazonBillingListener.java (hunk 2 above)
Comment (yurii-diachenko) [inline library/src/main/java/com/billing/amazon/AmazonBillingListener.java:76]: Yes, because thss info is too sensitive to be printed into production log

**Pull request merged**